### ISSUE TITLE
Test hangs forever with IntelliJ 2017.1

### ISSUE THREAD

**Issue by bluecmd** [bazelbuild/intellij#80]
Hi,

The issue I'm having is that when executing Bazel tests inside IntelliJ the tests never complete in the UI. The process exits just fine, but the UI doesn't update.

The IntelliJ log prints:
```
2017-04-26 13:49:04,415 [  55177]   INFO - .diagnostic.PerformanceWatcher - Unindexed files update took 4409ms; general responsiveness: ok; EDT responsiveness: ok 
2017-04-26 13:49:13,070 [  63832]  ERROR - ecution.process.ProcessHandler - null 
java.lang.reflect.InvocationTargetException
	at sun.reflect.GeneratedMethodAccessor156.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.execution.process.ProcessHandler$4.invoke(ProcessHandler.java:227)
	at com.sun.proxy.$Proxy16.processTerminated(Unknown Source)
	at com.intellij.execution.process.ProcessHandler$3.run(ProcessHandler.java:184)
	at com.intellij.execution.process.ProcessHandler$TasksRunner.execute(ProcessHandler.java:260)
	at com.intellij.execution.process.ProcessHandler.notifyTerminated(ProcessHandler.java:165)
	at com.intellij.execution.process.ProcessHandler.notifyProcessTerminated(ProcessHandler.java:161)
	at com.intellij.execution.process.KillableProcessHandler.notifyProcessTerminated(KillableProcessHandler.java:134)
	at com.intellij.execution.process.BaseOSProcessHandler.onOSProcessTerminated(BaseOSProcessHandler.java:187)
	at com.intellij.execution.process.OSProcessHandler.onOSProcessTerminated(OSProcessHandler.java:92)
	at com.intellij.execution.process.BaseOSProcessHandler$2$1.consume(BaseOSProcessHandler.java:152)
	at com.intellij.execution.process.BaseOSProcessHandler$2$1.consume(BaseOSProcessHandler.java:137)
	at com.intellij.execution.process.ProcessWaitFor$1.run(ProcessWaitFor.java:60)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NoSuchMethodError: com.intellij.execution.testframework.sm.runner.events.TestFinishedEvent.<init>(Ljava/lang/String;J)V
	at com.google.idea.blaze.base.run.smrunner.BlazeXmlToTestEventsConverter.processTestCase(BlazeXmlToTestEventsConverter.java:239)
	at com.google.idea.blaze.base.run.smrunner.BlazeXmlToTestEventsConverter.processTestSuite(BlazeXmlToTestEventsConverter.java:148)
	at com.google.idea.blaze.base.run.smrunner.BlazeXmlToTestEventsConverter.processTestSuite(BlazeXmlToTestEventsConverter.java:142)
	at com.google.idea.blaze.base.run.smrunner.BlazeXmlToTestEventsConverter.processTestSuites(BlazeXmlToTestEventsConverter.java:118)
	at com.google.idea.blaze.base.run.smrunner.BlazeXmlToTestEventsConverter.flushBufferBeforeTerminating(BlazeXmlToTestEventsConverter.java:97)
	at com.intellij.execution.testframework.sm.SMTestRunnerConnectionUtil$2.processTerminated(SMTestRunnerConnectionUtil.java:215)
	... 20 more
2017-04-26 13:49:13,085 [  63847]  ERROR - ecution.process.ProcessHandler - IntelliJ IDEA 2017.1.2  Build #IC-171.4249.39 
2017-04-26 13:49:13,085 [  63847]  ERROR - ecution.process.ProcessHandler - JDK: 1.8.0_112-release 
2017-04-26 13:49:13,086 [  63848]  ERROR - ecution.process.ProcessHandler - VM: OpenJDK 64-Bit Server VM 
2017-04-26 13:49:13,086 [  63848]  ERROR - ecution.process.ProcessHandler - Vendor: JetBrains s.r.o 
2017-04-26 13:49:13,086 [  63848]  ERROR - ecution.process.ProcessHandler - OS: Linux 
2017-04-26 13:49:13,086 [  63848]  ERROR - ecution.process.ProcessHandler - Last Action: RunClass 
```

I've traced it down to this code in `BlazeXmlToTestEventsConverter.java`:
```
[...]
  @Override
  public void flushBufferBeforeTerminating() {
    super.flushBufferBeforeTerminating();
    onStartTesting();
    getProcessor().onTestsReporterAttached();

    for (CompletedTestTarget testTarget : BlazeTestXmlFinderStrategy.locateTestXmlFiles(project)) {
      try (InputStream input = new FileInputStream(testTarget.testResultXml)) {
        parseXmlInput(getProcessor(), getKind(project, testTarget.label), input);
      } catch (Exception e) {
        // ignore parsing errors -- most common cause is user cancellation, which we can't easily
        // recognize.
      }
    }

[...]
  private void processTestCase(
      GeneralTestEventsProcessor processor, @Nullable Kind kind, TestSuite parent, TestCase test) {
[...]
    processor.onTestFinished(new TestFinishedEvent(displayName, parseTimeMillis(test.time)));
  }
```

What happens is that `onTestFinished` doesn't seem to be available with the signature that is expected, and the error is raised. As to why this incompatibility exists I have no idea, to my knowledge I'm using the same SDK as IntelliJ version so I'm figuring it should be the same signature that gets compiled. FWIW, I did find something that seemed similar in another project here: https://github.com/destin/maven-test-support-plugin/issues/14

My IntelliJ foo is too weak to be able to suggest a fix, but hopefully this is enough for someone to develop a fix.

**Comment by brendandouglas** [comment]
We haven't yet released a plugin version compatible with 2017.1.

We're in the process of rolling this out internally, and will release a Bazel plugin soon afterwards (scheduled for start of May sometime).

**Comment by brendandouglas** [comment]
The latest version (2017.04.17) has support for 2017.1, so this issue should no longer occur.
